Tests/DatadogTests/Datadog/URLSessionAutoInstrumentation/Interception/URLSessionInterceptorTests.swift

```diff
@@ -101,14 +94,77 @@ class URLSessionInterceptorTests: XCTestCase {
         )
 
         // Then
-        XCTAssertNotNil(interceptor.tracingHandler)
-        XCTAssertNotNil(interceptor.rumResourceHandler)
+        XCTAssertTrue(interceptor.handler is URLSessionRUMResourcesHandler)
+        XCTAssertTrue(
+            interceptor.injectTracingHeadersToFirstPartyRequests,
+            "Tracing headers should be injected when both Tracing and RUM instrumentations are enabled."
+        )
+        XCTAssertEqual(
+            interceptor.additionalHeadersForFirstPartyRequests,
+            [TracingHTTPHeaders.originField: TracingHTTPHeaders.rumOriginValue],
+            "Additional `x-datadog-origin: rum` header should be injected when both Tracing and RUM instrumentations are enabled."
+        )
     }
 
     // MARK: - URLRequest Interception
```

Comment:
we can add already traced request (_original request with tracing headers_) test case here
i guess we expect to have the original headers after the interception, correct?

Reply:
Sorry, I don't understand 🙂. In quoted lines we do not test requests, but only internal state of the `interceptor` (`interceptor.additionalHeadersForFirstPartyRequests`).

There's no such case, where `interceptor.modify(request:)` is called for a request, which was previously intercepted. This contract is guaranteed in swizzler tests.